src/rust/engine/process_execution/src/remote/streaming.rs

```diff
@@ -168,6 +168,10 @@ impl StreamingCommandRunner {
             build_id,
             &operation
           );
+
+          // Extract the operation name.
+          // Note: protobuf can return empty string for an empty field so convert empty strings
```

Comment:
IIRC, there is the equivalent of a "isset" method: ... maybe `has_*`?

Reply:
Not for primitive types where the field takes on the default value (empty string in this case). To your point, I had looked through the generated code for `Operation` and found the only methods related to `name` were: `clear_name`, `set_name`, `mut_name`, `take_name`, and `get_name`.

Reply:
And that occurs even if the field was marked optional:

```
If the default value is not specified for an optional element, a type-specific default value is used instead: for strings, the default value is the empty string. For bytes, the default value is the empty byte string. For bools, the default value is false. For numeric types, the default value is zero. For enums, the default value is the first value listed in the enum's type definition. This means care must be taken when adding a value to the beginning of an enum value list. See the Updating A Message Type section for guidelines on how to safely change definitions. 
```

https://developers.google.com/protocol-buffers/docs/proto#scalar

Reply:
Google actually has wrapper types for primitive value types to allow having the equivalent of `Some("")` and `None`.

Reply:
If this is just checking for "is_empty", the `trim` is a bit strange. But gotit!

Reply:
Just me protecting against whitespace. Probably overly cautious. I'm fine with removing it. Thoughts?